hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/server/namenode/ha/AbstractNNFailoverProxyProvider.java

```diff
@@ -200,6 +201,24 @@ public HAServiceState getCachedState() {
     return proxies;
   }
 
+  /**
+   * Resets the NameNode proxy addresses in case they're stale
+   */
+  protected void resetProxyAddresses(List<NNProxyInfo<T>> proxies) {
```

Comment:
instead of resetting all proxies, we can just reset the one currentProxyIndex pinted to after incrementing it, right?

Reply:
Yes this is a sensible suggestion. I'll work on incorporating that

Reply:
changed as suggested